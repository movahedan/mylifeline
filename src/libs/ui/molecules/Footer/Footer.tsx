import { footerContent } from "@mylifeline/contents/footer";
import { Text } from "@mylifeline/ui/atoms";
import { classNames } from "@mylifeline/utilities";

import type { CSSProperties, ReactNode } from "react";

export type FooterProps = {
  style?: CSSProperties;
  className?: string;
  children?: ReactNode;
};

export const Footer = ({ style, className, children }: FooterProps) => (
  <footer style={style} className={classNames([className])}>
    <Text as="p" className="mt-24 prose-badge">
      {footerContent.copyRight}
    </Text>

    {children}
  </footer>
);
