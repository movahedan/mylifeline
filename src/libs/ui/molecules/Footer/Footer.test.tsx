import { render } from "@testing-library/react";

import { Footer } from "./Footer";

describe("<Footer>", () => {
  it("should match is's snapshot", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot("default");
  });
});
