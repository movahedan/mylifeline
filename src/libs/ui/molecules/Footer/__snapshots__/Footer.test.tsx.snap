// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer> should match is's snapshot: default 1`] = `
<div>
  <footer
    class="footer w-full flex justify-center"
  >
    <p
      class="prose flex items-center prose-badge"
    >
      Created as a demo project Â© 2023.
    </p>
  </footer>
</div>
`;
